src/transformState.js

```diff
@@ -5,7 +5,27 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        for (const data in action.extraData) {
+          const prop = action.extraData[data];
+
+          state[data] = prop;
+        }
+        break;
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
```

Comment:
The variable 'key' does not provide clarity on what it holds. Consider renaming it to something more meaningful like 'propertyName'.